src/base/settingsstorage.h

```diff
@@ -62,4 +62,27 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class StorableValue
```

Comment:
Sorry, I was misleading in my previous code snippet, 
that doesn't work since `size().isValid()` returns `true` in member initializer list.

>But you can use it the following way: ... in constructor and then:

So this suggestion won't work too.

>Using pointer to CachedSettingValue breaks its goals (we can't use var = m_settingValue etc.).

you still can by dereferencing the pointer...
It's a bit ugly but this (`m_storeDialogSize`) is an special case , other class variables can use `CachedSettingValue` with no modifications.
IMO using a CachedSettingValue pointer is the easiest way now.